### PR TITLE
feat(get): add get

### PR DESCRIPTION
**Pull request by piquark6046**
# Summary
This PR addresses https://github.com/toss/es-toolkit/issues/215 and https://github.com/toss/es-toolkit/issues/91 by implementing the `get` function.

# Benchmark
Screenshot from 2024-07-18 17-11-36

### CODE REVIEW

**Comment by de-novo** [comment]
@piquark6046  I would like to suggest an improvement to the types in this PR. The current issue is that when the path depth exceeds one level, the return type is inferred as `unknown`. It would be beneficial if the return value of the `get` function could be clearly inferred.
스크린샷 2024-07-18 오후 7 39 21

Below is a type example that addresses this issue. This is a type I have used personally, and while it may not work perfectly in all situations, it should serve as a useful reference.

```ts
type Split<STR extends string, SEP extends string> = string extends SEP
  ? string[]
  : STR extends `${infer A}${SEP}${infer B}`
    ? [A, ...(B extends '' ? [] : Split<B, SEP>)]
    : SEP extends ''
      ? []
      : [STR];

type ReplaceAll<T extends string, A extends string, B extends string> = T extends `${infer C}${A}${D}`
  ? `${C}${B}${ReplaceAll<D, A, B>}`
  : T;

export type StringPathToArray<T extends string> = Split<ReplaceAll<ReplaceAll<T, '[', '.'>, ']', ''>, '.'>;

export type At<T extends object | any[], K extends Array<string | number>> = K extends [
  infer F extends string | number,
  ...infer R extends Array<string | number>,
]
  ? F extends keyof T
    ? T[F] extends object | any[]
      ? At<T[F], R>
      : T[F]
    : F extends `${infer N extends number}`
      ? N extends keyof T
        ? T[N] extends object | any[]
          ? At<T[N], R]
          : T[N]
        : never
      : never
  : T;

interface Test {
  a: { b: { c: number[] } };
}

type test = At<Test, ['a', 'b', 'c', 0]>; // number
```

**Review by raon0211 — APPROVED** [review]
Hey, thanks for your contribution! 

We were in the process of working on the `get` function. Let me combine our version with the one you implemented.

**Comment by raon0211** [comment]
Hello @de-novo , thanks for your suggestion!

Unfortunately, one of our core values of our library is [simple implementation](https://github.com/toss/es-toolkit/blob/main/.github/CONTRIBUTING.md#12-simplicity). I understand that your provided type helps our type safety, but I guess the type is too complicated to merge. We might open a new issue about this.

Let me merge this pull request, and let's discuss about correct types in the other issue if needed.

File: src/object/get.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * @returns {O[keyof O] | undefined} - The value from the object by the given path.
+ * 
+ * @example
+ * const obj = { a: { b: 3 } };
+ * get(obj, 'a.b', null); // 3
+ */
+
+export function get<O, T>(obj: O, path: string | string[], defaultValue?: T): O[keyof O] | T | undefined {
+  if (!Array.isArray(path)) {
+    let pathArr = path.split('.');
```
Comment (raon0211) [inline src/object/get.ts:21]:
```suggestion
    const pathArr = path.split('.');
```

**Review by raon0211 — APPROVED** [review]

**Pull request merged**